products/sentinel_one.py

```diff
@@ -163,6 +115,85 @@ def _authenticate(self):
             else:
                 raise ValueError(f'Error when authenticating to SentinelOne: {data}')
 
+    def _get_site_ids(self, site_id, account_id, account_name):
+        config = configparser.ConfigParser()
+        config.read(self.creds_file)
+
+        # instantiate site_ids and account_ids if not set
+        # make it all lists so more data can be appended from the config file
+        site_ids = (site_id) if site_id else list()
+        account_ids = (account_id) if account_id else list()
+        account_names = (account_name) if account_name else list()
+
+        # extract account/site ID from configuration if set
+        if 'account_id' in config[self.profile]:
+            for id in config[self.profile]['account_id'].split(','):
+                if id not in account_ids:
+                    account_ids.append(id.strip())
+
+        if 'site_id' in config[self.profile]:
+            for id in config[self.profile]['site_id'].split(','):
+                if id not in site_ids:
+                    site_ids.append(id.strip())
+
+        if 'account_name' in config[self.profile]:
+            for name in config[self.profile]['account_name'].split(','):
+                if name not in account_names:
+                    account_names.append(name.strip())
+
+        # determine site and account IDs to query (default is all)
+        self._site_ids = list()
+        self._account_ids = list()
+
+        if account_ids: # verify provided account IDs are valid
+            response = self._get_all_paginated_data(self._build_url(f'/web/api/v2.1/accounts'),
+                                                    params={'ids': ','.join(account_ids)},
+                                                    add_default_params=False)
+            for account in response:
+                if account['id'] not in self._account_ids:
+                    self._account_ids.append(account['id'])
+
+            diff = list(set(account_ids) - set(self._account_ids))
+            if len(diff) > 0:
+                self.log.warning(f'Account IDs {",".join(diff)} not found.')
+
+        temp_account_name = list()
+        for name in account_names: # verify provided account names are valid
+            response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/accounts'),
+                                                    params={'name': name},
+                                                    add_default_params=False)
+            for account in response:
+                temp_account_name.append(account['name'])
+                if account['id'] not in self._account_ids:
+                    self._account_ids.append(account['id'])
```

Comment:
```suggestion
                if account['name'] not in account_names:
                    account_names.append(account['name'])
```
should this be using self._account_names which doesn't appear to be defined at this point? Either way I believe this wasn't suppose to be checking accountId

Reply:
DV queries require account_ids so in this section, we're checking if we can resolve the provided account_names to an ID and then adding it to the accound_it list.

Reply:
I see what is occurring. this is fine how it is.